api4/system.go

```diff
@@ -515,3 +517,38 @@ func getRedirectLocation(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.MapToJson(m)))
 	return
 }
+
+func pushNotificationsAck(c *Context, w http.ResponseWriter, r *http.Request) {
+	ack := model.PushNotificationAckFromJson(r.Body)
+
+	if !*c.App.Config().EmailSettings.PushNotificationsAck {
+		c.Err = model.NewAppError("pushNotificationAck", "api.push_notifications_ack.disabled.app_error", nil, "", http.StatusForbidden)
+		return
+	}
+
+	var err *model.AppError
+	var notification *model.PushNotification
+	ackAt := model.GetMillis()
+
+	if pluginsEnvironment := c.App.GetPluginsEnvironment(); pluginsEnvironment != nil {
+		pluginContext := c.App.PluginContext()
+		pluginsEnvironment.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
+			notification, err = hooks.PushNotificationAck(pluginContext, ack.Id, ack.DeviceId, ack.ReceivedAt, ackAt)
+			return true
+		}, plugin.PushNotificationAckId)
+	}
+	if err != nil {
+		mlog.Debug(fmt.Sprintf("Push notification %s ack: message recived in the device %s at %d, and ack received at server at %d, but failed with the error %v", ack.Id, ack.DeviceId, ack.ReceivedAt, ackAt, err))
+		c.Err = err
+		return
+	}
+
+	mlog.Debug(fmt.Sprintf("Push notification %s ack: message recived in the device %s at %d, and ack received at server at %d", ack.Id, ack.DeviceId, ack.ReceivedAt, ackAt))
+
+	if notification != nil {
+		w.Write([]byte(notification.ToJson()))
```

Comment:
why do we need to return this in the response?

Reply:
It is not needed, but the idea is to be able to modify the notification during the ACK. But maybe is not the best way to do it.

Reply:
Just dropping the `return` seams fine, doesn't it?